ebi_eva_common_pyutils/reference/sequence.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2019 EMBL - European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import urllib
+from csv import DictReader, excel_tab
+from ftplib import FTP
+import re
+from urllib import request
+
+from cached_property import cached_property
+from retry import retry
+
+from ebi_eva_common_pyutils.command_utils import run_command_with_output
+from ebi_eva_common_pyutils.logger import AppLogger
+
+
+class NCBISequence(AppLogger):
+    """
+    Class that represent an Sequence that would originate from NCBI data
+    It takes a any Genbank or refseq accession and can download the sequence genomics fasta form.
+    Using species_scientific_name and assembly_accession it create a directory structure in the provided
+    reference_directory:
+        - species_scientific_name1
+            - assembly_accession1
+            - assembly_accession2
+        - species_scientific_name2
+    """
+
+    insdc_accession_formats = [
+        r'[A-Z][0-9]{5}\.[0-9]+',
+        r'[A-Z]{2}[0-9]{6}\.[0-9]+',
+        r'[A-Z]{2}[0-9]{8}\.[0-9]+',
+        r'[A-Z]{4}[0-9]{8}\.[0-9]+',
+        r'[A-Z]{6}[0-9]{9}\.[0-9]+'
+    ]
+
+    def __init__(self, sequence_accession, species_scientific_name, reference_directory, eutils_api_key=None):
+        self.sequence_accession = sequence_accession
+        self.species_scientific_name = species_scientific_name
+        self.reference_directory = reference_directory
+        self.eutils_api_key = eutils_api_key
+
+    @classmethod
+    def is_genbank_accession_format(cls, accession):
+        if any(
+                re.match(insdc_accession_format, accession)
+                for insdc_accession_format in cls.insdc_accession_formats
+        ):
```

Comment:
This isn't very important but you could do `'|'.join(insdc_accession_format)` and then just do a single regex match.

Reply:
I know but I find the `any` syntax more readable than the or and the cost of creating the different match object seem negligible.